doc/services/portability/posix/option_groups/index.rst

```diff
@@ -845,12 +845,14 @@ Enable this option with :kconfig:option:`CONFIG_POSIX_RAW_SOCKETS`.
 
 .. _posix_option_reader_writer_locks:
 
-_POSIX_READER_WRITER_LOCKS
+_POSIX_RW_LOCKS
 ++++++++++++++++++++++++++
 
-Enable this option with :kconfig:option:`CONFIG_POSIX_READER_WRITER_LOCKS`.
+Enable this option with :kconfig:option:`CONFIG_POSIX_RW_LOCKS`.
 
-.. csv-table:: _POSIX_READER_WRITER_LOCKS
+.. note:: POSIX_READER_WRITER_LOCKS and PTHREAD_RWLOCK are deprecated.
+
+.. csv-table:: _POSIX_RW_LOCKS
```

Comment:
This whole block should be moved to where the other Option Groups are located, right after [POSIX_PIPE](https://docs.zephyrproject.org/latest/services/portability/posix/option_groups/index.html#posix-pipe) to keep things in lexographic order.

Additionally, the POSIX Option Group is `POSIX_RW_LOCKS` (not leading underscore) and the POSIX Option is `_POSIX_READER_WRITER_LOCKS` (with leading underscore).

In this block, we want to change it to the Option Group rather than use the Option notation, and that would need to include the .rst anchor.